index.js

```diff
@@ -1,22 +1,60 @@
-module.exports = Hook
-
 var register = require('./lib/register')
 var addHook = require('./lib/add')
 var removeHook = require('./lib/remove')
 
-function Hook () {
+var bind = Function.bind
+var bindable = bind.bind(bind)
+
+function bindApi (hook, state, name) {
+  const removeHookRef = bindable(removeHook, null).apply(null, name ? [state, null, name] : [state, null])
```

Comment:
Can you explain the `bindable` hack here? I think it makes the code more complicated than it was before and I’m not quite sure why we need it?

Reply:
It's simple, we need to bind a variable amount of parameters, and this was one way of doing it:
https://stackoverflow.com/a/21792913

Reply:
Ahhh makes sense 👍 Could you add a comment with the stackoverlow link for future reference?

Reply:
Sure 😄